frontend/src/components/stat/chart/GenericChart.tsx

```diff
@@ -1,94 +1,104 @@
+                        x: {
+                            type: "category",
+                            labels: xLabels,
+                            ticks: { padding: 20 },
                         },
-                        scales: {
-                            x: {
-                                title: {
-                                    display: true,
-                                    text: chartXAxisLabel,
-                                },
-                            },
-                            y: {
-                                title: {
-                                    display: true,
-                                    text: chartYAxisLabel,
-                                },
-                            },
+                        y: {
+                            type: "category",
+                            labels: yLabels,
+                            ticks: { padding: 20 },
                         },
-                    }}
-                />
-            );
+                    },
+                },
+            ) as ChartOptions<"bubble">;
 
-        case "line":
             return (
-                <LineChart
+                <BubbleChart
                     data={{
-                        labels: Object.keys(chartData),
-                        datasets: [
-                            {
-                                label: props.legendName,
-                                data: Object.values(chartData),
-                            },
-                        ],
-                    }}
-                    options={{
-                        plugins: {
-                            title: {
-                                display: true,
-                                text: chartTitle,
-                            },
-                        },
-                        scales: {
-                            x: {
-                                title: {
-                                    display: true,
-                                    text: chartXAxisLabel,
-                                },
-                            },
-                            y: {
-                                title: {
-                                    display: true,
-                                    text: chartYAxisLabel,
-                                },
-                            },
-                        },
+                        datasets: [{ data: bubbleRows }],
                     }}
+                    options={bubbleChartOptions}
                 />
             );
 
+        case "histogram":
+            const histogramOptions = _.merge({}, chartOptions, {
+                x: { ticks: { padding: 20 } },
+            });
+            return (
+                <HistogramChart data={chartData} options={histogramOptions} />
+            );
+
         default:
-            return <p>yeah idk</p>;
+            return <p>You should not be able to see this</p>;
```

Comment:
If I don't have this, typescript complains that not all code paths resolve (cause it's a switch statement). Pretty annoying

Reply:
yeah i mean you could just throw an error here thats super generic because it should never get here